packages/@aws-cdk/region-settings/lib/settings/aws.ts

```diff
@@ -0,0 +1,97 @@
+import { Region } from '../region';
+
+// tslint:disable:variable-name Just replacing - with _ in region names...
+
+export const us_east_1 = new Region('us-east-1', {
```

Comment:
If we make this information come from a data (JSON) file, it'll be easy for users to hook additional data into there.

If it's all in code, not so much.

Reply:
I disagree here... If I go JSON:
- I'll now need to be schema-validating those (okay, fine)
- I can't model dynamic behavior using functions anymore
- Users won't be able to "hook additional data" there, because the file will be encapsulated in a package

Reply:
> I can't model dynamic behavior using functions anymore

Depends on what's in the JSON files. If it's just a lookup matrix, that's true. But it could be rules as well.

> Users won't be able to "hook additional data" there, because the file will be encapsulated in a package

Nobody says you can only ever load one file. You could load rules from multiple sources, and that's your extension point.